Userland/Libraries/LibCore/TempFile.cpp

```diff
@@ -1,57 +1,42 @@
 /*
- * Copyright (c) 2020-2021, the SerenityOS developers.
+ * Copyright (c) 2020-2023, the SerenityOS developers.
  *
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
 #include "TempFile.h"
-#include <AK/Random.h>
 #include <LibCore/DeprecatedFile.h>
-#include <fcntl.h>
-#include <stdlib.h>
-#include <sys/stat.h>
+#include <LibCore/System.h>
 
 namespace Core {
 
-NonnullOwnPtr<TempFile> TempFile::create(Type type)
+ErrorOr<NonnullOwnPtr<TempFile>> TempFile::create_temp_directory()
 {
-    return adopt_own(*new TempFile(type));
-}
+    char pattern[] = "/tmp/tmp.XXXXXX";
 
-DeprecatedString TempFile::create_temp(Type type)
-{
-    char name_template[] = "/tmp/tmp.XXXXXX";
-    switch (type) {
-    case Type::File: {
-        auto fd = mkstemp(name_template);
-        VERIFY(fd >= 0);
-        close(fd);
-        break;
-    }
-    case Type::Directory: {
-        auto fd = mkdtemp(name_template);
-        VERIFY(fd != nullptr);
-        break;
-    }
-    }
-    return DeprecatedString { name_template };
+    auto path = TRY(Core::System::mkdtemp(pattern));
+    return adopt_own(*new TempFile(Type::Directory, path));
 }
 
-TempFile::TempFile(Type type)
-    : m_type(type)
-    , m_path(create_temp(type))
+ErrorOr<NonnullOwnPtr<TempFile>> TempFile::create_temp_file()
 {
+    char file_path[] = "/tmp/tmp.XXXXXX";
+    TRY(Core::System::mkstemp(file_path));
+
+    auto string = TRY(String::from_utf8({ file_path, sizeof file_path - 1 }));
```

Comment:
This needs a comment. Why are we cutting off one character? And if we need to do that, can we the string returned by `mkstemp` instead?

Reply:
`mkstemp` returns a file descriptor, not a string

Reply:
ah, confused that with `mkdtemp` :)